clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/AmazonCertificateCachingAgent.groovy

```diff
@@ -76,29 +85,40 @@ class AmazonCertificateCachingAgent implements CachingAgent, AccountAware {
 
   @Override
   CacheResult loadData(ProviderCache providerCache) {
-    log.info("Describing items in ${agentType}")
-    def iam = amazonClientProvider.getAmazonIdentityManagement(account, region)
-
-    // Get all the target groups
-    List<ServerCertificateMetadata> iamCertificates = []
-    ListServerCertificatesRequest listServerCertificatesRequest = new ListServerCertificatesRequest()
-    while (true) {
-      def resp = iam.listServerCertificates(listServerCertificatesRequest)
-      iamCertificates.addAll(resp.serverCertificateMetadataList)
-      if (resp.marker) {
-        listServerCertificatesRequest.withMarker(resp.marker)
-      } else {
-        break
+    if (!lastFailure || lastFailure.isBefore(Instant.now().minus(1, ChronoUnit.HOURS))) {
+      log.info("Describing items in ${agentType}")
+      def iam = amazonClientProvider.getAmazonIdentityManagement(account, region)
+
+      // Get all the target groups
+      List<ServerCertificateMetadata> iamCertificates = []
+      ListServerCertificatesRequest listServerCertificatesRequest = new ListServerCertificatesRequest()
+      while (true) {
+        try {
+          def resp = iam.listServerCertificates(listServerCertificatesRequest)
+          iamCertificates.addAll(resp.serverCertificateMetadataList)
+          if (resp.marker) {
+            listServerCertificatesRequest.withMarker(resp.marker)
+          } else {
+            break
+          }
+        } catch (AmazonServiceException e) {
```

Comment:
as this stands, any exception calling the aws api is going to result in clearing the certificate list for an hour (on this instance)

I'd suggest: 
- trim the time down to something smaller (10 minutes maybe?)
- be more specific in the check on the AmazonServiceException to ensure it is actually a permission denied and not some other random service failure
- add a counter metric (`Registry.counter`) so that these can be surfaced to an alerting system that is not log based

Reply:
Thanks @cfieber. I've taken a shot at implementing all three suggestions. I'm unsure of the best practices for naming and tagging the metrics counter, so suggestions are more than welcome 👍

Reply:
@cfieber Can you please have another look at this?